site/en/blog/extension-instantnav/index.md

```diff
@@ -0,0 +1,179 @@
+of the outermost [frame is 0](/docs/extensions/reference/webNavigation/#a-note-about-frame-ids)
+may have problems. Since a tab can now have multiple outermost frames
+(prerendered and cached pages), the assumption that there is a single outermost
+frame for a Tab is incorrect. frameId == 0 will still continue to represent the
+outermost frame of the **active** page, but the outermost frames of **other**
+pages will be non-zero. A new field frameType has been added to fix this
+problem. See the [“How do I determine if a frame is the outermost frame?”](#outermost-frame)
+section.
+
+### Lifecycle of frame vs document
+
+Another concept that is problematic with extensions is the lifecycle of the
+frame. A frame hosts a document (which is associated with a committed URL).
+The document can change (say by navigating) but the *frameId* won’t, and so it
+is difficult to associate that something happened in a specific document with
+just *frameIds*. We are introducing a concept of a [documentId](/docs/extensions/reference/webNavigation/#method-getFrame:~:text=retrieve%20information%20about.-,documentId,-string%C2%A0optional)
+which is a unique identifier per document. If a frame is navigated and opens a
+new document the identifier will change. This field is useful for determining
+when pages change their lifecycle state (between prerender/active/cached)
+because it remains the same.
+
+### Web navigation events
+
+WebNavigation events can fire multiple times on the same page depending on the
+lifecycle it is in. See [“How do I tell what life cycle the page is in?”](#lifecyle)
+and [“How do I determine when a page transitions?”](#transitions) sections.
+
+## How do I tell what life cycle the page is in? {: #lifecyle}
+
+The [documentLifecycle](/docs/extensions/reference/extensionTypes/#type-DocumentLifecycle)
+attribute has been added to a number of extensions APIs where the *frameId* was
+previously available. If the value is present on an event (such as
+[onCommitted](/docs/extensions/reference/webNavigation/#event-onCommitted)),
+the value is the state at which the event was generated. You can always query
+information from the [WebNavigation GetFrame/GetAllFrames](/docs/extensions/reference/webNavigation/#method-getFrame)
+APIs, but using the value from the event is always preferred. If you do use the
+[GetFrame](/docs/extensions/reference/webNavigation/#method-getFrame) API be
+aware the state of the frame may have changed from when the event was generated
+to when the GetFrame promise is resolved.
+
+The [documentLifecycle](/docs/extensions/reference/extensionTypes/#type-DocumentLifecycle)
+has the following values:
+
+- **prerender** : Not currently presented to the user but preparing to possibly be displayed to the user.
+- **active**: Presently displayed to the user.
+- **cached**: Stored in the back/forward cache.
+- **pending_deletion**:  A special case where the document is actively being destroyed.
```

Comment:
Update formatting to better indicate these are strings that will be returned by the browser.

```suggestion
- `"prerender"`: Not currently presented to the user but preparing to possibly
  be displayed to the user.
- `"active"`: Presently displayed to the user.
- `"cached"`: Stored in the back/forward cache.
- `"pending_deletion"`:  A special case where the document is actively being
  destroyed.
```